src/basic_grasping_perception.cpp

```diff
@@ -58,8 +58,10 @@ class BasicGraspingPerception
     // use_debug: enable/disable output of a cloud containing object points
     nh_.getParam("use_debug", debug_);
 
+    // optionally enable object detection from the beginning without need to call the action
+    nh_.param<bool>("continuous_detection", continuous_detection_, false);
+
     // frame_id: frame to transform cloud to (should be XY horizontal)
-    world_frame_ = "base_link";
```

Comment:
So, I'm not sure how to make this backwards compatible - previously, if someone did not specify the "frame_id", they'd get a default of base_link, now they will get continuous detection (which could break some workflows). For instance, I know this would currently break the [Fetch gazebo-based demo](https://github.com/fetchrobotics/fetch_gazebo/blob/gazebo9/fetch_gazebo_demo/launch/demo.launch) which does not specify a base_link.

The "proper" way is probably to revert this line, then add a second parameter, something like "apply_transform_point_cloud", and default to true. If it were set to false, clear out the world_frame_ value.

Reply:
I am not sure what you mean with "continuous detection" here. The detection is not affected by the frame id, there is a separate parameter in this PR that allows a continuous detection without requiring actions, but this just sets the default frame name to the sensor frame (i.e. it skips the transformation).
I think this would only break where someone indeed would use `base_link` as a world frame.

Reply:
Sorry, I mixed up two thoughts here while reviewing.

Regardless, this is a change in behavior if someone has not manually set the "frame_id" and is relying on the default parameter - and will likely break their use case (there is a check inside the segmentation that the plane is "horizontal" - which makes the assumption that the fixed frame be planar with the ground - the sensor frame is rarely so.

While I can link the above Fetch demo (which relies on the default parameter), I don't know how many other users out there are using the default parameter. I suspect is most users since the best known demonstration of this system is the Fetch project I linked.

Reply:
Ok, should I then just remove this specific commit?

Reply:
Do you have a use case that actually involves not-transforming to a fixed frame? If you want to support that use case, that'd be fine - but I think it has to be implemented as a second parameter as noted in my first comment. If you don't actually have that use case, then yeah, might as well just remove the commit.

Reply:
The idea was to provide the detection directly in the sensor frame so that it can be used without a robot kinematic chain. In such a case, we could skip the transformation entirely. But I think this actually breaks as the support plane is expected to be perpendicular to the z-axis (or x-y-plane parallel). This would likely not work with a camera sensor frame anyway.